substrate/utils/frame/benchmarking-cli/src/overhead/cmd.rs

```diff
@@ -18,29 +18,73 @@
 //! Contains the [`OverheadCmd`] as entry point for the CLI to execute
 //! the *overhead* benchmarks.
 
-use sc_block_builder::BlockBuilderApi;
-use sc_cli::{CliConfiguration, ImportParams, Result, SharedParams};
-use sc_client_api::UsageProvider;
-use sc_service::Configuration;
-use sp_api::{ApiExt, CallApiAt, ProvideRuntimeApi};
-use sp_runtime::{traits::Block as BlockT, DigestItem, OpaqueExtrinsic};
-
-use clap::{Args, Parser};
-use log::info;
-use serde::Serialize;
-use std::{fmt::Debug, path::PathBuf, sync::Arc};
-
+use super::runtime_utilities::*;
 use crate::{
 	extrinsic::{
 		bench::{Benchmark, BenchmarkParams as ExtrinsicBenchmarkParams},
 		ExtrinsicBuilder,
 	},
-	overhead::template::TemplateData,
-	shared::{HostInfoParams, WeightParams},
+	overhead::{
+		cmd::ChainType::{Parachain, Relaychain, Unknown},
+		fake_runtime_api,
+		template::TemplateData,
+	},
+	shared::{
+		genesis_state,
+		genesis_state::{GenesisStateHandler, SpecGenesisSource},
+		HostInfoParams, WeightParams,
+	},
+};
+use clap::{ArgGroup, Args, Parser};
+use codec::Encode;
+use cumulus_client_parachain_inherent::MockValidationDataInherentDataProvider;
+use fake_runtime_api::RuntimeApi as FakeRuntimeApi;
+use frame_support::Deserialize;
+use genesis_state::WARN_SPEC_GENESIS_CTOR;
+use log::info;
+use polkadot_parachain_primitives::primitives::Id as ParaId;
+use sc_block_builder::BlockBuilderApi;
+use sc_chain_spec::{ChainSpec, ChainSpecExtension, GenesisBlockBuilder};
+use sc_cli::{CliConfiguration, Database, ImportParams, Result, SharedParams};
+use sc_client_api::{execution_extensions::ExecutionExtensions, UsageProvider};
+use sc_client_db::{BlocksPruning, DatabaseSettings};
+use sc_executor::WasmExecutor;
+use sc_service::{new_client, new_db_backend, BasePath, ClientConfig, TFullClient, TaskManager};
+use serde::Serialize;
+use serde_json::{json, Value};
+use sp_api::{ApiExt, CallApiAt, Core, ProvideRuntimeApi};
+use sp_blockchain::HeaderBackend;
+use sp_core::H256;
+use sp_inherents::{InherentData, InherentDataProvider};
+use sp_runtime::{
+	generic,
+	traits::{BlakeTwo256, Block as BlockT},
+	DigestItem, OpaqueExtrinsic,
+};
+use sp_storage::Storage;
+use sp_wasm_interface::HostFunctions;
+use std::{
+	fmt::{Debug, Display, Formatter},
+	fs,
+	path::PathBuf,
+	sync::Arc,
 };
+use subxt::{client::RuntimeVersion, ext::futures, Metadata};
+
+const DEFAULT_PARA_ID: u32 = 100;
+const LOG_TARGET: &'static str = "polkadot_sdk_frame::benchmark::overhead";
 
 /// Benchmark the execution overhead per-block and per-extrinsic.
 #[derive(Debug, Parser)]
+#[clap(group(
+    ArgGroup::new("genesis_source")
+        .args(["chain", "runtime"])
+        .required(true)
+), group(
+    ArgGroup::new("chain_source")
```

Comment:
This group is only used to defined a conflict further down. Its a bit annoying, but I did not find a proper way to reference the `chain` command from the flattened `SharedParams` in any other way.